server/acks_utils.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright © 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+// Helper functions to be called from request handlers defined in the source
+// file acks_handlers.go.
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	httputils "github.com/RedHatInsights/insights-operator-utils/http"
+	"github.com/RedHatInsights/insights-operator-utils/types"
+	ira_server "github.com/RedHatInsights/insights-results-aggregator/server"
+)
+
+// readRuleSelectorAndJustificationFromBody function tries to read data
+// structure types.AcknowledgementRuleSelectorJustification from response
+// payload (body)
+func readRuleSelectorAndJustificationFromBody(writer http.ResponseWriter, request *http.Request) (
+	types.AcknowledgementRuleSelectorJustification, error) {
+
+	// try to read request body
+	var parameters types.AcknowledgementRuleSelectorJustification
+	err := json.NewDecoder(request.Body).Decode(&parameters)
+
+	if err != nil {
+		log.Error().Err(err).Msg("wrong payload provided by client")
+		// return HTTP code 400 to client
+		http.Error(writer, err.Error(), http.StatusBadRequest)
+		return parameters, err
+	}
+
+	// everything seems to be ok
+	return parameters, nil
+}
+
+// readOrgIDAndUserIDFromToken helper method reads organization ID and user ID
+// (account number) from the token
+func (server *HTTPServer) readOrgIDAndUserIDFromToken(writer http.ResponseWriter, request *http.Request) (
+	types.OrgID, types.UserID, error) {
+	// auth. token contains organization ID and user ID we need to use
+	authToken, err := server.GetAuthToken(request)
+	if err != nil {
+		handleServerError(writer, err)
+		return types.OrgID(0), types.UserID(""), err
+	}
+
+	// Organization ID and user ID are to be provided in the token
+	orgID := authToken.Internal.OrgID
+	userID := authToken.AccountNumber
+	return orgID, userID, nil
+}
```

Comment:
I think that we already have some functions to extract the user and organization IDs